front/src/applications/editor/tools/rangeEdition/components/TrackRange.tsx

```diff
@@ -0,0 +1,179 @@
+                extremity: 'END',
+              },
+            });
+          }}
+          onMouseLeave={() => setState({ hoveredItem: null })}
+          onMouseEnter={() =>
+            setState({
+              hoveredItem: {
+                itemType: 'TrackRangeExtremity',
+                track: trackState.track,
+                position: getPointAt(trackState.track, range.end),
+                extremity: 'END',
+              },
+            })
+          }
+        >
+          <FaFlagCheckered />
+        </button>
+      </small>
+      <small>
+        <button
+          type="button"
+          className="btn btn-primary btn-sm px-2"
+          aria-label={t('common.delete')}
+          title={t('common.delete')}
+          onClick={() => {
+            const newEntity = cloneDeep(entity);
+            newEntity.properties.track_ranges?.splice(index, 1);
+            setState({ entity: newEntity, hoveredItem: null });
+          }}
+          onMouseLeave={() => setState({ hoveredItem: null })}
+          onMouseEnter={() =>
+            setState({
+              hoveredItem: {
+                itemType: 'TrackRange',
+                track: trackState.track,
+                position: getPointAt(trackState.track, trackState.track.properties.length / 2),
+              },
+            })
+          }
+        >
+          <FaTimes />
+        </button>
+      </small>
+    </div>
+  );
+
+  const applicableDirectionSelect = (
+    <select
+      id="filterLevel"
+      className="form-control"
+      value={range.applicable_directions}
+      onChange={(e) => {
+        const newEntity = cloneDeep(entity);
+        const newRange = (newEntity.properties.track_ranges || [])[index];
+        newRange.applicable_directions = e.target.value as ApplicableDirection;
+        setState({ entity: newEntity, hoveredItem: null });
+      }}
+    >
+      {APPLICABLE_DIRECTIONS.map((direction) => (
+        <option key={direction} value={direction}>
+          {t(`Editor.directions.${direction}`)}
+        </option>
+      ))}
+    </select>
+  );
+  const applicableDirection =
+    entityIsSpeedSection && !speedRestrictionTool
+      ? applicableDirectionSelect
+      : t(`Editor.directions.${range.applicable_directions}`);
+  return (
+    <li
+      key={`track-range-${range.track}-${range.begin}-${range.end}-${range.applicable_directions}-${index}`}
+      className="mb-4 d-flex flex-row align-items-center"
+    >
+      {(!trackState || trackState.type === 'loading') && (
+        <div className="position-relative w-100" style={{ height: 50 }}>
+          <LoaderFill className="bg-transparent" />
+        </div>
+      )}
+      {trackState?.type === 'success' && (
+        <>
+          {buttons}
+          <div className="flex-grow-1 flex-shrink-1 ml-3">
+            <EntitySumUp entity={trackState.track} />
+            {applicableDirection}
+          </div>
+        </>
+      )}
+    </li>
+  );
+};
```

Comment:
Separate the returns :
```suggestion
  if (!trackState || trackState.type === 'loading') {
    return (
      <li key={key} className="mb-4 d-flex flex-row align-items-center">
        <div className="position-relative w-100" style={{ height: 50 }}>
          <LoaderFill className="bg-transparent" />
        </div>
      </li>
    );
  }
  
  if (trackState.type === 'error') return null;

  return (
    <li key={key} className="mb-4 d-flex flex-row align-items-center">
      {!speedRestrictionTool && (
        <TrackRangeButtons index={index} range={range} track={trackState.track} />
      )}
      <div className="flex-grow-1 flex-shrink-1 ml-3">
        <EntitySumUp entity={trackState.track} />
        {entityIsSpeedSection && !speedRestrictionTool ? (
          <ApplicableDirectionSelect index={index} range={range} />
        ) : (
          t(`Editor.directions.${range.applicable_directions}`)
        )}
      </div>
    </li>
  );
```

Reply:
really good improvement